File: section1.2/dualpal.java


import java.io.*;
import java.math.BigInteger;

/*
ID: jameszh1
LANG: JAVA
TASK: dualpal
*/
public class dualpal {
    public static void main (String[] args) throws Exception{
        BufferedReader br = new BufferedReader(new FileReader("dualpal.in"));
        PrintWriter pw = new PrintWriter(new BufferedWriter(new FileWriter("dualpal.out")));

        String[] input = br.readLine().split(" ");
        int number = Integer.parseInt(input[0]);
        int start = Integer.parseInt(input[1]);
        int i = start + 1;
        while (number > 0) {
            BigInteger b = new BigInteger(String.valueOf(i), 10);
            int count = 0;
            for (int base = 2; base <= 10; base++) {
                String s = b.toString(base);
                if (isPalindromic(s) && number > 0) {
                    count++;
                    if (count >= 2) {
                        pw.println(b);
                        number--;
                        break;
                    }
                }
            }
            i++;
        }

        pw.close();
        br.close();
    }

    public static boolean isPalindromic(String s) {
        StringBuffer sb = new StringBuffer(s);
        if (sb.toString().equals(sb.reverse().toString())) return true;
        else return false;
    }
}

File: Python/chapter1/section3/dualpal.py
"""
ID: jameszh1
LANG: PYTHON2
TASK: dualpal
"""
def is_palindromic(string):
    return string == string[::-1]

def baseN(num,b):
  return ((num == 0) and  "0" ) or ( baseN(num // b, b).lstrip("0") + "0123456789abcdefghijklmnopqrstuvwxyz"[num % b])

fin = open ('dualpal.in', 'r')
fout = open ('dualpal.out', 'w')
N, S = map(int, fin.readline().split())
start = S + 1
while(N > 0):
    count = 0
    for base in range(2, 11):
        if (is_palindromic(str(baseN(start, base)))):
            count = count + 1
            if (count >= 2):
                fout.write(str(start) + '\n')
                N = N - 1
                break
    start = start + 1

fout.close()